core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -39,6 +40,47 @@
 @ThreadSafe
 public final class UnderFileSystemManager {
   /**
+   * An object which generates input streams to under file system files given a position. This
+   * class does not manage the life cycles of the generated streams and it is up to the caller to
+   * close the streams when appropriate.
+   */
+  private final class UnderFileSystemInputStream {
```

Comment:
I think `Agent` is a better term, updated. I've marked it as not thread safe (and should only be used by 1 thread at a time).